File: Stack/ValidateStackSequence.java
/*https://leetcode.com/problems/validate-stack-sequences/*/

class Solution {
    public boolean validateStackSequences(int[] pushed, int[] popped) {
        Stack stack = new Stack();
        int popIndex = 0;
        for (int i = 0; i < pushed.length; ++i)
        {
            stack.push(pushed[i]);
            while (popIndex < popped.length && !stack.isEmpty() && popped[popIndex] == (Integer)stack.peek())
            {
                stack.pop();
                ++popIndex;
            }
        }
        return stack.isEmpty();
    }
}

File: Stack/validate_stack_sequence.py
# https://leetcode.com/problems/validate-stack-sequences/submissions/

class Solution:
    def validateStackSequences(self, pushed: List[int], popped: List[int]) -> bool:
        stack = list()
        pop_index = 0;
        for i in range(len(pushed)):
            stack.append(pushed[i])
            while (pop_index < len(popped) and len(stack) > 0 and popped[pop_index] == stack[len(stack)-1]):
                stack.pop()
                pop_index += 1
        return len(stack) == 0